tasks/main.yml

```diff
@@ -5,23 +5,148 @@
 - name: Set platform/version specific variables
   include_tasks: tasks/set_vars.yml
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Check if requested version is supported in the system (RHEL8)
+  fail:
+    msg: "RHEL 8 supports only Postgresql 10, 12 and 13"
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "8"
+    - postgresql_version not in rhel8_streams
+
+- name: Check if requested version is supported in system (RHEL9)
+  fail:
+    msg: "RHEL 9 supports only Postgresql 13"
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "9"
+    - postgresql_version not in rhel9_streams
+
+- name: Check requested and installed version of Postgresql
+  fail:
+    msg: "Setting version {{ postgresql_version }} while
+      {{ ansible_facts.packages['postgresql'][0].version }} is installed"
+  when:
+    - "'postgresql' in ansible_facts.packages"
+    - ansible_facts.packages['postgresql'][0].version |
+      regex_search('([0-9]*).', '\\1') | first !=  postgresql_version
+
+- name: Ensure required packages are installed on RHEL8 based systems
   package:
-    name: "{{ __template_packages }}"
+    name: "@postgresql:{{ postgresql_version }}/server"
     state: present
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "8"
+
+# RHEL 9 doesn't support modules yet
+- name: Ensure required packages are installed on RHEL9 based systems and others
+  package:
+    name: postgresql-server
+    state: present
+  when:
+    - ansible_facts['distribution_major_version'] != "8"
```

Comment:
What about Fedora?  Does the role support EL7?  Note that this is why we have ` - name: Set platform/version specific variables` in set_vars.yml - so that you can define `vars/main.yml` with
```yaml
__postgresql_packages: [postgresql-server]
```
Then define `vars/RedHat_8.yml` and `vars/CentOS_8.yml` with
```
__postgresql_packages: ["@postgresql:{{ postgresql_version }}/server"]
```
So by default the name of the postgresql package will be `postgresql-server`, but on EL8 it will be overridden to be `"@postgresql:{{ postgresql_version }}/server"`

then in tasks/main.yml you can have simply
```
- name: Ensure postgresql packages
  package:
    name: "{{ __postgresql_packages }}"
```